op-chain-ops/script/script.go

```diff
@@ -377,6 +392,8 @@ func (h *Host) onExit(depth int, output []byte, gasUsed uint64, err error, rever
 			h.log.Warn("Revert", "addr", addr, "err", err, "revertData", hexutil.Bytes(output), "depth", depth)
 		}
 	}
+
+	h.callStack[len(h.callStack)-1].GasUsed += gasUsed
```

Comment:
The risky thing here about gas-tracking is the warm/cold storage difference. In script execution we are in the same tx context throughout. I believe forge recently introduced a special feature where top-level broadcast calls can be turned into individual tx-contexts, to effectively reset the warm/cold situation, and get accurate gas accounting. I don't think we're event using that yet (since no parallel tx submission), but we should. I'll look into it, it is probably just a simple call to get the tx context changed, but geth might be weird.

Reply:
Fix: https://github.com/ethereum-optimism/optimism/pull/11839

Reply:
merged 11839 in here